plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-attributes-controller.php

```diff
@@ -24,4 +24,87 @@ class WC_REST_Product_Attributes_Controller extends WC_REST_Product_Attributes_V
 	 * @var string
 	 */
 	protected $namespace = 'wc/v3';
+
+	/**
+	 * Generates a unique slug for a given attribute name. We do this so that we can 
+	 * create more than one attribute with the same name.
+	 *
+	 * @param string $attribute_name The attribute name to generate a slug for.
+	 */
+	private function generate_unique_slug( $attribute_name ) {
+		global $wpdb;
+
+		$root_slug = wc_sanitize_taxonomy_name( $attribute_name );
+
+		$results = $wpdb->get_results(
+			$wpdb->prepare( "SELECT attribute_name FROM {$wpdb->prefix}woocommerce_attribute_taxonomies WHERE attribute_name LIKE %s ORDER BY attribute_id DESC LIMIT 1", $root_slug . '%' )
+		);
+
+		// The slug is already unique!
+		if ( empty( $results ) ) {
+			return $root_slug;
+		}
+
+		$last_created_slug = $results[0]->attribute_name;
+		$suffix = intval( substr( $last_created_slug, strrpos( $last_created_slug, '-' ) + 1 ) );
+
+		return $root_slug . '-' . ( $suffix + 1 );
+	}
+
+	/**
+	 * Create a single attribute.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Request|WP_Error
+	 */
+	public function create_item( $request ) {
+		global $wpdb;
+
+		$generate_slug = stripslashes( $request['generate_slug'] );
+		$slug          = wc_sanitize_taxonomy_name( stripslashes( $request['slug'] ) );
+
+		if ( ! empty( $generate_slug ) ) {
```

Comment:
```suggestion
		if ( ! empty( $generate_slug ) && $generate_slug === 'true'  ) {
```
The `empty` tag check if something is `false` but the `generate_slug` will be a string, so it doesn't adhere correctly if `false` is actually passed in.

Reply:
done -> https://github.com/woocommerce/woocommerce/pull/39827/commits/efd59de7f2a29242c29b71cb9645ca6e7f560058